### PR TITLE
feat: distro stuff and pacstall 5.2.0 vars for srcinfo

### PR DESCRIPTION
**Pull request by oklopfer**

### CODE REVIEW

File: pacup [hunk 1]
```diff
@@ -457,17 +457,39 @@ sub main ($pkg) {
     my $pkgdir = tempdir "$pkgname.XXXXXX", DIR => $PACUP_DIR;
     my ( undef, $ppath_tmp ) = tempfile "$pacscript.XXXXXX", DIR => $pkgdir;
     writelines( \@plines, $ppath_tmp );
-
+    my $listfile = "distrolist";
+    my $distrolist_cont;
+    if ( -f $listfile ) {
+        open my $fh, '<', $listfile or throw "Cannot open $listfile: $!";
+        $distrolist_cont = do { local $/; <$fh> };
+        close $fh;
```
Comment (ghost) [inline pacup]:
```suggestion
        $distrolist_cont = do {
            local $/;
            open my $fh, '<', $listfile or throw "Cannot open $listfile: $!";
            <$fh>;
        };
```

File: pacup [hunk 2]
```diff
@@ -457,17 +457,39 @@ sub main ($pkg) {
     my $pkgdir = tempdir "$pkgname.XXXXXX", DIR => $PACUP_DIR;
     my ( undef, $ppath_tmp ) = tempfile "$pacscript.XXXXXX", DIR => $pkgdir;
     writelines( \@plines, $ppath_tmp );
-
+    my $listfile = "distrolist";
```
Comment (ghost) [inline pacup]: Make this / resolve this into an absolute path

**Review by ghost — CHANGES_REQUESTED** [review]

File: pacup (hunk 2 above)
Comment (oklopfer) [reply pacup]: it’s meant to be relative though

**Review by ghost — APPROVED** [review]

File: pacup [hunk 3]
```diff
@@ -457,17 +457,41 @@ sub main ($pkg) {
     my $pkgdir = tempdir "$pkgname.XXXXXX", DIR => $PACUP_DIR;
     my ( undef, $ppath_tmp ) = tempfile "$pacscript.XXXXXX", DIR => $pkgdir;
     writelines( \@plines, $ppath_tmp );
-
+    my $listfile = $origcwd . "distrolist";
```
Comment (vigress8) [inline pacup:460]: Needs a slash

Comment (ghost) [inline pacup]:
```suggestion
    my $listfile = $origcwd . "/distrolist";
```

**Pull request merged**